spec/requests/company_spec.rb

```diff
@@ -0,0 +1,279 @@
+require 'rails_helper'
+
+RSpec.describe Company, type: :request do
+  let(:user) { create(:user, :admin) }
+  let(:headers) { { "Authorization": sign_in(user) } }
+
+  describe 'GET /index' do
+    subject(:fetch_companies) do
+      get companies_path, headers: headers
+    end
+
+    let!(:companies) { create_list(:company, 2) }
+
+    it 'returns the correct 200 response code' do
+      fetch_companies
+
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'returns the expected response body' do
+      fetch_companies
+
+      expect(json).to eq(serialize_all(companies.sort_by(&:name)))
+    end
+  end
+
+  describe 'GET /show' do
+    subject(:fetch_company) do
+      get company_path(target_company), headers: headers
+    end
+
+    context 'when the company exists' do
+      let(:target_company) { create(:company) }
+
+      it 'returns the correct 200 response code' do
+        fetch_company
+
+        expect(response).to have_http_status(:ok)
+      end
+
+      it 'returns the expected response body' do
+        fetch_company
+
+        expect(json).to eq(serialize(target_company))
+      end
+    end
+
+    context 'when company is not found' do
```

Comment:
I think it's not necessary

Reply:
I think it is since we have a specific response to this situation and we have the same tests in user specs.

Reply:
OK, talking about request context, it make sense.